connectivity/drivers/ble/FEATURE_BLE/COMPONENT_CYW43XXX/CyH4TransportDriver.cpp

```diff
@@ -236,6 +236,13 @@ void CyH4TransportDriver::terminate()
 #else
     cyhal_uart_free(&uart);
 #endif
+
+    // Unlock Deep Sleep if it was locked by CyH4TransportDriver before.
+    if (bt_host_wake_active == true)
+    {
+       sleep_manager_unlock_deep_sleep();
+       bt_host_wake_active = false;
+    }
```

Comment:
`host_wake_pin` should also be deleted to not leak memory when `initialize` is called again and prevent toggle of deep sleep lock.

Reply:
Hi @pan, 

The `host_wake_pin ` is deleted in CyH4TransportDriver::terminate() line#226. Regarding prevent the toggling of deep sleep lock in `initialize` , do you mean add something like:
```
CyH4TransportDriver::CyH4TransportDriver(...)
{
   ...
    bt_host_wake_active = false;
   ...
}

void CyH4TransportDriver::initialize()
{
 ...
    if (!bt_host_wake_active)
    {
         bt_host_wake_active = true;
         sleep_manager_lock_deep_sleep();
     }
 ...
}
```

Regards,
Nazar

Reply:
I missed the `delete`. Sorry about that. My comment about toggling of deep sleep lock was in the case where the in is not deleted as the irq handlers would still be called. 

All good for me. I mark it as resolved.